extra/mariabackup/xtrabackup.cc

```diff
@@ -3441,14 +3441,12 @@ static bool xtrabackup_copy_mmap_logfile()
 
   for (unsigned retry_count{0};;)
   {
-    recv_sys_t::parse_mtr_result r;
     const byte *start= &log_sys.buf[recv_sys.offset];
-
-    if (recv_sys.parse_mmap<recv_sys_t::store::BACKUP>(false) ==
-        recv_sys_t::OK)
+    recv_sys_t::parse_mtr_result r=
+        recv_sys.parse_mmap<recv_sys_t::store::BACKUP>(false);
+    if (r == recv_sys_t::OK)
     {
       const byte *end;
-
```

Comment:
As far as I can tell, we do not need any local variable `r` here. The only possible return values of `parse_mmap<recv_sys_t::store::BACKUP>()` are `recv_sys_t::OK` and `recv_sys_t::GOT_EOF`. The return value `recv_sys_t:::PREMATURE_EOF` is only possible in `recv_sys_t::parse_mtr()`, and the return value `recv_sys_t::GOT_OOM` is only possible when either function template is instantiated for the template parameter `recv_sys_t::store::YES`.